framework/encode/vulkan_capture_manager.cpp

```diff
@@ -715,14 +715,24 @@ VkResult VulkanCaptureManager::OverrideCreateBuffer(VkDevice
                                                     const VkAllocationCallbacks* pAllocator,
                                                     VkBuffer*                    pBuffer)
 {
-    VkResult result           = VK_SUCCESS;
-    auto     device_wrapper   = reinterpret_cast<DeviceWrapper*>(device);
-    VkDevice device_unwrapped = device_wrapper->handle;
-    auto     device_table     = GetDeviceTable(device);
+    VkResult                  result                = VK_SUCCESS;
+    auto                      device_wrapper        = reinterpret_cast<DeviceWrapper*>(device);
+    VkDevice                  device_unwrapped      = device_wrapper->handle;
+    auto                      device_table          = GetDeviceTable(device);
+    auto                      handle_unwrap_memory  = VulkanCaptureManager::Get()->GetHandleUnwrapMemory();
+    const VkBufferCreateInfo* pCreateInfo_unwrapped = UnwrapStructPtrHandles(pCreateInfo, handle_unwrap_memory);
+
+    VkBufferCreateInfo modified_create_info = (*pCreateInfo_unwrapped);
+
+    if (IsTrimEnable())
+    {
+        modified_create_info.usage |= VK_BUFFER_USAGE_TRANSFER_SRC_BIT;
```

Comment:
Can this always succeed or could there be `BufferCreateInfo`s that will become un-creatable if `USAGE_TRANSFER_SRC` is added?

Reply:
I think it should be able  always succeed.

Reply:
According to this website, no devices support `protectedMemory`, so I can't test `VK_BUFFER_CREATE_PROTECTED_BIT`.
http://www.vulkan.gpuinfo.org/listfeaturescore11.php

Also, I read `vkspec`, it didn't mention about that `VK_BUFFER_CREATE_PROTECTED_BIT can't use with USAGE_TRANSFER_SRC`. VkImage didn't, either. So, I think we don't need to worry about `protectedMemory`.